core/metrics/src/main/java/org/phenotips/remote/metrics/internal/providers/NumberOfVariantsProvider.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses/
+ */
+package org.phenotips.remote.metrics.internal.providers;
+
+import org.phenotips.remote.metrics.spi.MetricProvider;
+
+import org.xwiki.component.annotation.Component;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.hibernate.Query;
+import org.hibernate.Session;
+
+import com.xpn.xwiki.store.hibernate.HibernateSessionFactory;
+
+/**
+ * Computes the number of variant objects associated with genes that are candidate or solved that are available for
+ * matching using MME in this node. This includes duplicates.
+ *
+ * @version $Id$
+ */
+@Component
+@Named("numberOfVariants")
+@Singleton
+public class NumberOfVariantsProvider implements MetricProvider
+{
+    @Inject
+    private HibernateSessionFactory sessionFactory;
+
+    @Override
+    public Object compute()
+    {
+        Session session = null;
+        try {
+            session = this.sessionFactory.getSessionFactory().openSession();
+            Query q = session.createQuery("select count (variantObj.name) from "
+                + HQL_BASE_MME_PATIENT_FILTER_FROM
+                + ", BaseObject geneObj, BaseObject variantObj, StringProperty geneStatusProp"
+                + " where "
+                + HQL_BASE_MME_PATIENT_FILTER_WHERE
+                + " and geneObj.name = doc.fullName and geneObj.className = 'PhenoTips.GeneClass'"
+                + " and variantObj.name = doc.fullName and variantObj.className = 'PhenoTips.GeneVariantClass'"
+                + " and geneStatusProp.id.id = geneObj.id and geneStatusProp.id.name = 'status'"
+                + " and geneStatusProp.value in ('candidate', 'solved')");
```

Comment:
This is missing the connection between the variant object and the gene object, which means it may contain variants for rejected genes. The query should be just like for `NumberOfUniqueVariantsProvider`, except that the `distinct` should be missing from the `select` part.

Reply:
corrected in next commit, which i forgot to add to PR, but it is included in preprod jar.